### DISCUSSION TITLE
HA Critical Error - Received mismatched identity for device Smart Plug

### DISCUSSION THREAD

**Discussion by boba51** [krahabb/meross_lan#511] [Q&A]
Hi,

I'm after some assistance with this issue, as I've Googled the error and searched this form with no success and would like to try and resolve this issue.

I'm fairly new to Home Assistant although have tried it a few times in the past without great success, currently installed on a VM running on my Synology NAS and of course have number of Meross Smartplugs hence my use of this integration.

After installing the integration, I had an issue that others had configuring the devices, but following the fix all my smartplugs discovered and were working fine, but following a reboot of the NAS (I can't remember if it was planned or a a loss of power), I now get the Topic line error msg constantly in HA.

All my Meross devices appear to be functioning OK, including the device with the IP Address given when I click on the error msg.

All help appreciated, please bear in mind although IT literate, newbie when it comes to HA and so any requests for logs may need a basic guide to obtain it/them, also please advise if I need to remove any details form said logs which I shouldn't publicly share publicly. 

From memory as it was a little while ago now, to try and resolve the issue I deleted the device with the above IP Address and it got rediscovered but the error msg remains :(

Thanks

Bob

**Comment by krahabb** [comment]
Hello @boba51,
It appears as if your plugs are 'swapping' ip addresses due to DHCP. This in turn will 'confuse' meross_lan which is anyway able to detect the device communicating is different than configured.
The 'critical' is because this situation might be 'dangerous' because you think you'r maybe operating a plug  with HA while instead the command is going to another one.
Usually, when devices update or swap addresses due to DHCP renewal, HA and then meross_lan should be able to detect the change and update the configured ip for the device but if, for any reason, the DHCP update is not processed by HA/meross_lan, this is the outcome.

I suggest you to either 'fix' the DHCP address assignment policy or better check if the HA instance in receiving DHCP updates. At any rate, even if DHCP updates are correctly received/processed by HA, there could be some cases where the system fails (think when HA is not yet 'online' after a reset/reboot and devices swap addresses)

So in the end, the only reliable policy is to have the router to always assign the same address to devices.

**Reply by boba51** [reply]
Hi,

Thanks, I assumed this was the case and am already in the process of reserving IP Addresses for all my devices, but as my ASUS Router needs to reboot when setting reservations (why I don't know) I'm working around my families Internet needs.

Will this error msg disappear on its own or is there something I can do to reset the device/s to get rid of it ?

Thanks,

Bob

**Reply by krahabb** [reply]
well..You have maybe to reload the configuration for a particular entry once you're sure the ip address matches correctly

**Reply by boba51** [reply]
Thanks, anyway to identify the device from error and locate it in HA to reload it?

Sorry if this is obvious, I haven't been using HA for very long so everything a little new to me.

Bob

**Reply by boba51** [reply]
Hi,

I've reserved IP Addresses for all the devices on my network, I've been through every Meross device in under Meross LAN, configured them all successfully and rebooted HA, unfortunately the error msg has reoccurred.

I'm still wondering if there's a way to pinpoint the device causing the issue and remove it from the configuration and let it get rediscovered?

Thanks

Bob

**Comment by boba51** [comment]
Issue resolved, thank you @krahabb 
The problem device was offline so Meross LAN couldn't find it, I plugged it in (3rd one I tried) reserved it's new IP Address and rebooted HA - Error finally gone 👍
